_cite/plugins/orcid.py

```diff
@@ -33,36 +33,67 @@ def query(_id):
     # list of sources to return
     sources = []
 
-    # go through response structure and pull out ids e.g. doi:1234/56789
+    # filter id by some criteria. return true to accept, false to reject.
+    def filter_id(_id):
+        # is id of certain "relationship" type
+        relationships = ["self", "version-of", "part-of"]
+        if not get_safe(_id, "external-id-relationship", "") in relationships:
+            return False
+
+        id_type = get_safe(_id, "external-id-type", "")
+
+        # is id of certain type
+        # types = ["doi"]
+        # if id_type not in types:
+        #     return False
+
+        # is id citable by manubot
+        if id_type not in manubot_citable:
+            return False
+
+        return True
+
+    # prefer some ids over others by some criteria. return lower number to prefer more.
+    def sort_id(_id):
+        id_type = get_safe(_id, "external-id-type", "")
+        types = [
+            "doi",
+            # "arxiv",
+            # "url",
+        ]
```

Comment:
I was going to suggest reading this from `_config.yaml`, but it looks like that's just for Jekyll configuration. If you can add custom entries to that config file, it might be nice to read this list from there and default to `["doi"]` if not found, but obviously not a requirement. (IIRC a commentor [#327](https://github.com/greenelab/lab-website-template/issues/327) suggested that.)

Reply:
I try to reserve the config file for only the most high-level and commonly configured things, to avoid the inner-platform effect and having a hundred different options. For things like this, we expect the user to just edit the template directly. There's already a few customizable behaviors at this same level that are done in the cite code as commented out suggestions.